src/backend/distributed/transaction/remote_transaction.c

```diff
@@ -65,6 +69,102 @@ static char *IsolationLevelName[] = {
 };
 
 
+PG_FUNCTION_INFO_V1(citus_internal_start_management_transaction);
+PG_FUNCTION_INFO_V1(execute_command_on_other_nodes);
+PG_FUNCTION_INFO_V1(citus_mark_object_distributed);
+PG_FUNCTION_INFO_V1(commit_management_command_2PC);
+
+bool IsManagementCommand = false;
+FullTransactionId outerXid;
+static MultiConnection *managementCon = NULL;
+bool CitusManagementDatabase = true;
```

Comment:
Do we need this boolean? Seems cheap enough to just check the database name in IsCitusManagementDatabase

Reply:
I cannot use `CurrentDatabaseName()` function in the post commit hook, where I need the `IsCitusMainDb()` check. Because it needs to be in a transaction to be able to access the database names.

Reply:
One way to check if the database you are connected is the MainDb or not would be to add the following change:

> diff --git a/src/backend/distributed/shared_library_init.c b/src/backend/distributed/shared_library_init.c
> index 5b0208eca..bbf0d2bc7 100644
> --- a/src/backend/distributed/shared_library_init.c
> +++ b/src/backend/distributed/shared_library_init.c
> @@ -3139,6 +3139,11 @@ CitusAuthHook(Port *port, int status)
>         InitializeBackendData(port->application_name);
> 
> 
> **+       bool isInMainDB = strcmp(port->database_name, MainDb) == 0;**
>         /* let other authentication hooks to kick in first */
>         if (original_client_auth_hook)
>         {

Note that:

- Need to make **isInMainDB**  global
- CitusAuthHook is run for all client connections. But it  is not run for the background workers (such as  autovacuum launcher, Citus maintenance deamon). So you need to initialize isInMainDB  with a default value such as isInMainDB  = false; 

`CoordinatedTransactionCallback `will be called in those background workers as well. There will be a maintenance deamon connected to MainDb. If we need to do 2PC in there, we should set isMainDB to true there.